client/components/app.jsx

```diff
@@ -2,16 +2,42 @@ import React from 'react';
 import GradeTable from './grade-table';
 import Header from './header';
 import GradeForm from './grade-form';
+
 export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.getAverageGrade = this.getAverageGrade.bind(this);
     this.addGrade = this.addGrade.bind(this);
+    this.deleteGrade = this.deleteGrade.bind(this);
+
     this.state = {
       grades: []
     };
   }
 
+  deleteGrade(deleteId) {
+
+    fetch(`api/grades/${deleteId}`, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }
+    ).then(res => res.json())
+      .then(data => {
+        const deleteGrade = this.state.grades.slice();
+        const targetIndex = deleteGrade.findIndex(grade => grade.id === deleteId);
+        deleteGrade[targetIndex] = data;
```

Comment:
you can skip assigning the response body at the index, since you'll be splicing it out anyway.